src/features/staking/stCELO/StakeForm.tsx

```diff
@@ -0,0 +1,258 @@
+  disabled?: boolean;
+}) {
+  const { stakingRate, unstakingRate } = useExchangeRates();
+  const maxAmountWei = useMemo(
+    () =>
+      (getMaxAmount(action, stCELOBalances?.usable, walletBalance) || 0n) - MIN_REMAINING_BALANCE,
+    [action, stCELOBalances, walletBalance],
+  );
+  const rate = action === LiquidStakeActionType.Stake ? stakingRate : unstakingRate;
+  const { setFieldValue, values } = useFormikContext<LiquidStakeFormValues>();
+  useEffect(() => {
+    setFieldValue('amount', Math.max(0, fromWei(maxAmountWei - MIN_REMAINING_BALANCE)));
+  }, [maxAmountWei, setFieldValue]);
+
+  return (
+    <div>
+      <AmountField
+        maxValueWei={maxAmountWei}
+        maxDescription={`${action === LiquidStakeActionType.Stake ? '' : 'st'}CELO available`}
+        disabled={disabled}
+      />
+      <div className="flex items-center justify-between pt-4">
+        <label className="pl-0.5 text-xs font-medium">Receive</label>
+        <strong className="text-xs">
+          {rate != null ? formatNumberString(values.amount / rate, 5) : 'Loading exchange rate…'}{' '}
+          {action === LiquidStakeActionType.Unstake ? '' : 'st'}CELO
+        </strong>
+      </div>
+      <div className="flex items-center justify-between">
+        <label className="pl-0.5 text-xs font-medium">Exchange Rate</label>
+        <span className="text-xs">
+          {action === LiquidStakeActionType.Stake ? stakingRate : unstakingRate}
+        </span>
+      </div>
+    </div>
+  );
+}
+
+function validateForm(
+  values: LiquidStakeFormValues,
+  stCELOBalances: ReturnType<typeof useStCELOBalance>['stCELOBalances'],
+  walletBalance: bigint,
+): FormikErrors<LiquidStakeFormValues> {
+  const { action, amount } = values;
+
+  // TODO implement toWeiAdjusted() and use it here
+  const amountWei = toWei(amount);
+  if (!amountWei || amountWei <= 0n) return { amount: 'Invalid amount' };
+
+  const maxAmountWei = getMaxAmount(action, stCELOBalances.usable, walletBalance);
+  if (!maxAmountWei || amountWei > maxAmountWei) {
+    return { amount: 'Amount exceeds max' };
+  }
+
+  // Special case handling for locking whole balance
+  if (action === LiquidStakeActionType.Stake) {
+    const remainingAfterPending = amountWei - stCELOBalances.usable;
+    if (walletBalance - remainingAfterPending <= MIN_REMAINING_BALANCE) {
+      return { amount: 'Cannot lock entire balance' };
+    }
+  }
+
+  // Ensure user isn't trying to unlock CELO used for staking
+  if (action === LiquidStakeActionType.Unstake) {
+    if (stCELOBalances.lockedVote > amountWei) {
+      toast.warn(
+        'Locked funds that have voted for a governance cannot be unlocked until the proposal is resolved.',
+      );
+      return { amount: 'Locked funds in use' };
+    }
+
+    const nonVotingBalance = stCELOBalances.usable;
+    if (amountWei > nonVotingBalance) {
+      toast.warn(
+        'Locked funds that are current staked must be unstaked before they can be unlocked.',
```

Comment:
Based on the code the error should be more like

Amount to UnStake must not be greater than the unstakable stCelo. stCelo voting on proposals is locked while proposal is in flight.